airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java

```diff
@@ -610,11 +617,22 @@ public String getJobMainContainerMemoryLimit() {
     return getEnvOrDefault(JOB_MAIN_CONTAINER_MEMORY_LIMIT, DEFAULT_JOB_MEMORY_REQUIREMENT);
   }
 
+  /**
+   * There are two types of environment variables available to the job container:
+   * <ul>
+   * <li>Exclusive variables prefixed with JOB_DEFAULT_ENV_PREFIX</li>
+   * <li>Shared variables defined in JOB_SHARED_ENVS</li>
+   * </ul>
+   */
   @Override
   public Map<String, String> getJobDefaultEnvMap() {
-    return getAllEnvKeys.get().stream()
+    final Map<String, String> jobPrefixedEnvMap = getAllEnvKeys.get().stream()
         .filter(key -> key.startsWith(JOB_DEFAULT_ENV_PREFIX))
         .collect(Collectors.toMap(key -> key.replace(JOB_DEFAULT_ENV_PREFIX, ""), getEnv));
+    final Map<String, String> jobSharedEnvMap = JOB_SHARED_ENVS.entrySet().stream().collect(Collectors.toMap(
+        Entry::getKey,
+        entry -> Objects.requireNonNullElse(entry.getValue().apply(this), "")));
```

Comment:
I'd like to not set a default empty value for two reasons. Firstly the consumer needs to remember to check this default value, which can be forgotten relatively easily. Secondly when the consumer does not check for the default value, their behavior may be broken. For example, in the future, we will pass in the Sentry DSN through an env variable from this class. When the DSN is `null` or empty, it knows to not do any tracking. If anyone forgets to check for the default value when constructing a Sentry client, the client will fail because of an invalid DSN.